File: api-code-examples/go/blob-list-incomplete-blobs.go
package main

import (
	"fmt"

	"github.com/n0-computer/iroh-ffi/iroh-go/iroh"
)

func main() {
	node, err := iroh.NewIrohNode("iroh_data_dir")
	if err != nil {
		panic(err)
	}

	// typically only happens if you have not finished syncing or interrupted
	// a download
	incompletes, err := node.BlobsListIncomplete()
	if err != nil {
		panic(err)
	}

	fmt.Println("Incomplete blobs:")
	for _, res := range incompletes {
		fmt.Printf("\thash: %s size: %s expected size: %d\n", res.Hash.ToString(), res.Size, res.ExpectedSize)
	}

	// Output:
	// Incomplete blobs:
}

File: api-code-examples/python/blob-list-incomplete-blobs.py
import iroh
import asyncio

async def main():
    # Create in memory iroh node
    node = await iroh.IrohNode.memory()

    # Typically only happens if you have not finished syncing or interrupted
    # a download
    incompletes = await node.blobs_list_incomplete()

    print("Incomplete blobs:")
    for res in incompletes:
        print(f"\thash: {res.hash} size: {res.size} expected size: {res.expected_size}")

# Output:
# Incomplete blobs:

asyncio.run(main())
